pkg/handler/eventhandler.go

```diff
@@ -83,9 +84,73 @@ type TypedEventHandler[object any, request comparable] interface {
 }
 
 var _ EventHandler = Funcs{}
+var _ EventHandler = TypedFuncs[client.Object, reconcile.Request]{}
+var _ TypedEventHandler[client.Object, reconcile.Request] = TypedFuncs[client.Object, reconcile.Request]{}
 
 // Funcs implements eventhandler.
-type Funcs = TypedFuncs[client.Object, reconcile.Request]
+type Funcs = funcs[client.Object, reconcile.Request]
+
+// funcs implements eventhandler
+type funcs[object client.Object, request reconcile.Request] struct{}
```

Comment:
Actually this is not correct, right now this is just a re-implementation of `EnqueueRequestForOwner` 😅 We need to have public properties for CreateFunc/UpdateFunc/DeleteFunc/GenericFunc and delegate to them just like we do in `TypedFuncs`

Reply:
Does `Funcs` need to be it's own type? I can't add methods to `Funcs` if it's a `TypedFuncs[client.Object, reconcile.Request]` and declaring `CreateFunc` methods  causes an overflow of calls if I try to inject it in the `h.CreateFunc(ctx, e, q)` call.  🤔 

Do you want the default behavior of the `priorityqueue`  to be based on if the `CreateFunc` is not defined?

Reply:
> Do you want the default behavior of the priorityqueue to be based on if the CreateFunc is not defined?

I think if the `CreateFunc` is not defined we should just not do anything just like we do today.

Maybe a better approach is to use `Reflect` in `TypedFuncs` to figure out if `object` is assignable to `client.Object` and `request` is a `reconcile.Request` and if so, use the helpers you added for `EnqueueRequestForObject`? I did some of that in the `Builder` because `For` there can only be used if `request` is a `reconcile.Request`